### ISSUE TITLE
Peer info during handshake

### ISSUE THREAD

**Issue by jmozah** [ethersphere/bee#1351]
It would be very useful to applications like crawler to have the following info during peer connect/handshake
- OS
- bee version 
- uptime
- db-capacity (configured/ used)

Of-course this can also be a protocol after connection. Also it can be like an optional one so that people who dont want to share this info with crawler can block it.

**Comment by Eknir** [comment]
Note that all values can be lied about. So, you could never trust this

**Comment by Eknir** [comment]
I would personally not convolute the discovery with this, but can imagine it would make sense as a separate protocol

**Comment by jmozah** [comment]
> Note that all values can be lied about. So, you could never trust this

Yes. I agree. But so is every other information you get from a peer in any of the protocols (ex: connected peers in gossip).


> I would personally not convolute the discovery with this, but can imagine it would make sense as a separate protocol

Agree. I have the same thinking. May be it should be a separate protocol with options to enable or disable during node startup. So that it is optional for users to report this to his peers. very similar to '--ethstats' flag in go-ethereum.

With this we would get a very important stat from the network (total disk capacity of the network). Even some approximation is a good thing to have.

Let me know of this is okay and I can PR this sometime if you guys have no issues.

**Comment by Eknir** [comment]
Hi there,

Thanks for thinking with us on this. If you want to estimate the total disk size, then you can just take the default disk size and multiply that with the number of nodes. This would yield a more reliable result than basing your estimate on what peers report to you because:

1) peers may lie
2) even when 1 peer is having 10tb of storage on Swarm, this does not influence the actual capacity of Swarm (see https://docs.ethswarm.org/docs/advanced/large-node-operators).

Besides, any new protocol adds a larger attack surface. 

Because of this, I don't believe we should work on this.
